### PR TITLE
icon-loader: match more application icons via fallbacks

### PR DESCRIPTION
**Pull request by jlindgren90**
Add fallbacks to:

1) match reverse-URL .desktop file names when the app_id is just the
   base name (e.g. "thunderbird" -> org.mozilla.Thunderbird.desktop)

~2) look in /usr/share/pixmaps when the icon isn't present in the
   hicolor theme (still the case for many applications)~

### CODE REVIEW

**Comment by vyivel** [comment]
> look in /usr/share/pixmaps when the icon isn't present in the hicolor theme

libsfdo is supposed to do this already.

**Comment by jlindgren90** [comment]
> libsfdo is supposed to do this already.

Hmm, it is not working for me (tested with asunder, xterm, xmaxima...)

File: src/icon-loader.c [hunk 1]
```diff
@@ -135,6 +137,33 @@ process_abs_name(struct icon_ctx *ctx, const char *icon_name)
...
+	struct sfdo_desktop_entry **entries = sfdo_desktop_db_get_entries(db, &n_entries);
+
+	for (size_t i = 0; i < n_entries; i++) {
+		struct sfdo_desktop_entry *entry = entries[i];
+		const char *desktop_id = sfdo_desktop_entry_get_id(entry, NULL);
+		/* Get portion of desktop ID after last '.' */
+		const char *dot = strrchr(desktop_id, '.');
+		const char *desktop_id_base = dot ? (dot + 1) : desktop_id;
+
+		if (g_ascii_strcasecmp(app_id, desktop_id_base) == 0) {
```
Comment (johanmalm) [inline src/icon-loader.c:159]: Prefer `!strcasecmp()`

**Review by johanmalm — COMMENTED** [review]
LGTM. 
Just one preference to use `strcasecmp()` rather than the glib variant.

File: src/icon-loader.c (hunk 1 above)
Comment (jlindgren90) [reply src/icon-loader.c:159]: My concerns with strcasecmp is the locale-specific behavior - it is "implementation defined" in all but the C/POSIX locale. But maybe it's not an issue in practice.

Comment (johanmalm) [reply src/icon-loader.c]: Okay. Let's leave as is.

Comment (johanmalm) [reply src/icon-loader.c]: Ah... sorry you've changed it :smile:

Comment (jlindgren90) [reply src/icon-loader.c]: Oh, I already switched it to strcasecmp. Either way is fine with me.

**Comment by johanmalm** [comment]
Thanks

**Pull request merged**